Update Datadog SDKs from `v4` => `v5`

nmajor25 opened the pull request:
Shortcut Story ID: [sc-42902]

nmajor25 committed:
Update datadog packages from v4 => v5

nmajor25 committed:
Replace deprecated sdk init parameter - allowedTracingOrigins => allowedTracingUrls

nmajor25 committed:
Remove 'trackFrustrations: true' parameter

By setting 'trackUserInteractions: true', user frustration signals are automatically tracked

Reference: https://docs.datadoghq.com/real_user_monitoring/guide/browser-sdk-upgrade/#frustration-signals-are-collected-automatically

nmajor25 committed:
Explicitily set 'trackResources: true'

Reference: https://docs.datadoghq.com/real_user_monitoring/guide/browser-sdk-upgrade/#resources-and-long-task-tracking

nmajor25 commented on src/js/datadog.js:
Pertaining to this: https://docs.datadoghq.com/real_user_monitoring/guide/browser-sdk-upgrade/#automatic-recording-of-sessions-sampled-for-session-replay

Does `datadogRum.startSessionReplayRecording();` need to be called in the `setUser()` function? Or can we remove that line of code and have it automatically record at the beginning of the session?

But maybe we need to keep that there until after the `setUser()` is called? I’m not sure.

nmajor25 replied:
If we remove `startSessionReplayRecording()`, we’ll need to set `sessionReplaySampleRate: 100`. The default for that changed from `100` => `0`. ([reference](https://docs.datadoghq.com/real_user_monitoring/guide/browser-sdk-upgrade/#default-session-replay-sampling-rate))  

Otherwise, it’d be set to `0` by default and no replays would be recorded.

nmajor25 replied:
If we need to keep using `startSessionReplayRecording()`, we’ll need to set the `startSessionReplayRecordingManually: true` init parameter.

paulfalgout replied:
I think setting the manual would be ok.  There's very very little the user can do prior to bootstrap that we could glean from watching a session replay.  That way we're also only watching the session for a logged in user.

nmajor25 replied:
Ya that makes sense. I'll work on getting that updated. 

And I'll go through the change log again and make sure we're all good in terms of handling any other breaking changes.

nmajor25 replied:
Okay I just added the `startSessionReplayRecordingManually: true` init option.

Since we use `startSessionReplayRecording()`, my understanding is that we don't need to set `sessionReplaySampleRate: 100`. That's only used if the sessions recordings are automatically started when the session begins (without `startSessionReplayRecording()`)

nmajor25 commented on src/js/datadog.js:
There were also some small changes to the `beforeSend()` method. But I don't think they effect anything for us.

nmajor25 committed:
Set 'startSessionReplayRecordingManually: true' init option

Since we customize when a session recording starts via 'startSessionReplayRecording() after the user is signed in, v5 requires that we set 'startSessionReplayRecordingManually: true'

Reference: https://docs.datadoghq.com/real_user_monitoring/guide/browser-sdk-upgrade/#automatic-recording-of-sessions-sampled-for-session-replay
